connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceTaskTest.java

```diff
@@ -256,39 +277,76 @@ public void testPartitionStateMutation() {
         partitionStates.put(sourceTopicPartition, partitionState);
         RecordMetadata recordMetadata = new RecordMetadata(sourceTopicPartition, metadataOffset, 0, 0, 0, recordPartition);
 
-        when(outstandingOffsetSyncs.tryAcquire()).thenReturn(true);
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(final InvocationOnMock invocation) {
+                final Callback callback = invocation.getArgument(1);
+                callback.onCompletion(null, null);
+                return null;
+            }
+        }).when(producer).send(any(), any());
         mirrorSourceTask.commitRecord(sourceRecord, recordMetadata);
         assertEquals(recordOffset, partitionState.lastSyncUpstreamOffset, "sync offsets");
         assertEquals(metadataOffset, partitionState.lastSyncDownstreamOffset, "sync offsets");
         assertEquals(recordOffset, partitionState.previousUpstreamOffset, "sync offsets");
         assertEquals(metadataOffset, partitionState.previousDownstreamOffset, "sync offsets");
-        assertFalse(partitionState.shouldSyncOffsets);
+        assertFalse(partitionState.shouldSyncOffsets, "partition state reset");
+        verify(producer, times(1)).send(any(), any());
 
-        int newRecordOffset = 2;
-        int newMetadataOffset = 102;
-        recordMetadata = new RecordMetadata(sourceTopicPartition, newMetadataOffset, 0, 0, 0, recordPartition);
+        recordOffset = 2;
+        metadataOffset = 102;
+        recordMetadata = new RecordMetadata(sourceTopicPartition, metadataOffset, 0, 0, 0, recordPartition);
         sourceRecord = mirrorSourceTask.convertRecord(new ConsumerRecord<>(topicName, recordPartition,
-                newRecordOffset, System.currentTimeMillis(), TimestampType.CREATE_TIME, recordKey.length,
+                recordOffset, System.currentTimeMillis(), TimestampType.CREATE_TIME, recordKey.length,
                 recordValue.length, recordKey, recordValue, headers, Optional.empty()));
 
-        when(outstandingOffsetSyncs.tryAcquire()).thenReturn(false);
+        // Do not release outstanding sync semaphore
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(final InvocationOnMock invocation) {
+                return null;
+            }
+        }).when(producer).send(any(), any());
```

Comment:
Using the `ArgumentCaptor` approach:
```suggestion
        // Do not immediately release the outstanding sync semaphore
        doReturn(null).when(producer).send(any(), producerCallback.capture());
```

Reply:
Updated.